packit/api.py

```diff
@@ -732,6 +734,43 @@ def sync_status_string(
 """
         return f"'{source_git}' is up to date with '{dist_git}'."
 
+    def check_version_distance(
+        self,
+        current,
+        proposed,
+        target_branch,
+    ) -> None:
+        """Following this guidelines:
+        https://docs.fedoraproject.org/en-US/fesco/Updates_Policy/#philosophy
+        we want to avoid major updates of packages within a **stable** release.
+
+        current: str, already released version for package
+        proposed: str, release we are preparing for package
+        target_branch: str, Fedora branch where release the package
+        """
+        if target_branch in get_branches("fedora-branched"):
+            masked_current = re.match(self.package_config.version_update_mask, current)
+            masked_proposed = re.match(
+                self.package_config.version_update_mask,
+                proposed,
+            )
+            if (
+                masked_current
+                and masked_proposed
+                and masked_current.group(0) != masked_proposed.group(0)
+            ):
+                raise ReleaseSkippedPackitException(
+                    f"The upstream released version {current} does not match "
+                    f"specfile version {proposed} at branch {target_branch} "
+                    f"using the default version_update_mask "
+                    f'"{self.package_config.version_update_mask}".'
+                    "\nPackit wants to avoid major updates of packages within a stable release."
```

Comment:
I would change the wording a bit to be more generic since people can use this with some other mask.